server/py/services/api/crud/pipelines.py

```diff
@@ -89,10 +90,21 @@ def list_pipelines(
             filter_=filter_,
             experiment_ids=experiment_ids,
         )
-
-        runs, next_page_token = self._paginate_runs(
-            kfp_client, page_token, page_size, sort_by, filter_json
-        )
+        try:
+            runs, next_page_token = self._paginate_runs(
+                kfp_client, page_token, page_size, sort_by, filter_json
+            )
+        except MLRunBadRequestError as e:
+            if "no support for filtering on unrecognized field" in str(e):
+                mlrun.utils.logger.warning(
+                    "KFP 1.8 does not support experiment filtering, retrying without it",
+                )
+                filter_json = mlrun.utils.get_kfp_list_runs_filter(
+                    filter_=filter_,
+                )
+                runs, next_page_token = self._paginate_runs(
+                    kfp_client, page_token, page_size, sort_by, filter_json
+                )
```

Comment:
I would expect a better approach to know the kfp server version ahead of time, perhaps when MLRun starts (or initiates the config) it can get/sample the kfp server version - in here we can use that to determine the code flow